packages/nimble-components/src/table-column/base/index.ts

```diff
@@ -30,6 +30,9 @@ export abstract class TableColumn<
     @attr({ attribute: 'sort-direction' })
     public sortDirection: TableColumnSortDirection = TableColumnSortDirection.none;
 
+    @attr({ attribute: 'sorting-disabled', mode: 'boolean' })
+    public sortingDisabled = false;
```

Comment:
I vaguely recall a discussion where we thought we should move all the sorting code to a mixin class, similar to [`groupable-column.ts`](https://github.com/ni/nimble/blob/main/packages/nimble-components/src/table-column/mixins/groupable-column.ts). Does anyone else remember that discussion better than I do? @atmgrifter00 @rajsite 

If we do want it as a mixin, would it make sense to do that as part of this PR or as a separate refactor?

Reply:
I believe we would only move it to a mixin if we felt it made sense that a column type itself would never be sortable. At the moment, I don't believe we have come up with such a column. Admittedly, it's unclear if we will ever have a column that is intrinsically "ungroupable". A DateTime column would be the closest thing I can think of, but even that is questionable.